x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/user_menu_links.tsx

```diff
@@ -48,5 +62,20 @@ export const createUserMenuLinks = (cloud: CloudStart): UserMenuLink[] => {
     });
   }
 
+  userMenuLinks.push({
+    content: (
+      <ThemDarkModeToggle
+        useUpdateUserProfile={useUpdateUserProfile}
+        getSpaceDarkModeValue={() => {
+          return uiSettingsClient.get('theme:darkMode');
+        }}
```

Comment:
My initial comment was:

> Should we really retrieve the darkMode value from the uiSettings client? Having the getting using the uiSettings while the setter uses `useUpdateUserProfile` from the security plugin feels wrong?  How is this value propagated from the profile service to the uiSettings client exactly?

When, I saw the implementation of the `ThemDarkModeToggle` component and I understood how the implementation works.

However, I feel like there may be a bad isolation of concerns here tbh. I would expect `ThemDarkModeToggle` to just don't know how this value is coming from, and just accepts a getter and a setter for `darkMode` (so basically moves most of the logic from the `ThemDarkModeToggle` component to `createUserMenuLinks`). 

It would feel more "natural" to me, wdyt?

That may be a NIT though.

Reply:
Oh you mean a "smart" and a "dumb" component? 😄 

I see what you mean. At the same time the trade off is the the userMenuLinks start to know "a lot".

I'd prefer one of those options:

- A connected component and a UI component (the one you mention)
- Instead of the connected component have a hook with that logic (I think I'll go that route WDYT?)

Reply:
I moved the connected logic to a hook (https://github.com/elastic/kibana/pull/161214/commits/c792b65e43bdae6aa1e31ec4fc3661cbafc18473)